### PR TITLE
Scottx611x/cleanup tool containers

### PR DESCRIPTION
**Pull request by scottx611x**
Add `post_delete` signal to handle removal of Tool Docker containers

### CODE REVIEW

File: refinery/tool_manager/tests.py [hunk 1]
```diff
@@ -1169,6 +1170,33 @@ def test_bad_POST_transaction_rollback(self):
...
+            tool_annotation = [json.loads(f.read())]
+
+        with mock.patch(self.mock_vis_annotations_reference,
+                        return_value=tool_annotation):
+            call_command("generate_tool_definitions", visualizations=True)
+
+        td = ToolDefinition.objects.all()[0]
+        post_data = {
+            "tool_definition_uuid": td.uuid,
+            "file_relationships": "['www.coffee.com/cool_file.txt']"
```
Comment (hackdna) [inline refinery/tool_manager/tests.py]: Should be a IANA-managed Reserved Domain: https://github.com/refinery-platform/refinery-platform/wiki/Copy-Style-Guide#miscellaneous

File: refinery/tool_manager/models.py [hunk 2]
```diff
@@ -305,6 +307,19 @@ def update_file_relationships_string(self):
         self.save()
 
 
+@receiver(post_delete, sender=Tool)
```
Comment (hackdna) [inline refinery/tool_manager/models.py]: If container is created after tool instance, I'd make it a pre_delete handler, so container would be stopped before tool instance is deleted to preserve logical order.

File: refinery/tool_manager/tests.py [hunk 3]
```diff
@@ -1192,12 +1192,10 @@ def test_tool_container_removed_on_deletion(self):
         self.view(post_request)
 
         tool = Tool.objects.get(tool_definition__uuid=td.uuid)
+        tool_uuid = tool.uuid
```
Comment (hackdna) [inline refinery/tool_manager/tests.py]: Why is `tool_uuid` variable necessary?

Comment (scottx611x) [inline refinery/tool_manager/tests.py]: When I delete the Tool I lose access to its members, and I need its UUID for the DockerClientWrapper call below

Comment (hackdna) [inline refinery/tool_manager/tests.py]: I see. Actually, there is no need to test external libraries: `DockerClientWrapper().purge_by_label()` in `remove_tool_container()` should be mocked and then tested if it was called then there is no need to save the tool UUID.

Comment (scottx611x) [inline refinery/tool_manager/tests.py]: If I mock the purging of the container, then its never actually purged in the test

**Pull request merged**